libraries/AP_OSD/AP_OSD_Screen.cpp

```diff
@@ -189,6 +200,19 @@ void AP_OSD_Screen::draw_gspeed(uint8_t x, uint8_t y)
     backend->write(x, y, false, "%3.0f%c", v, SYM_KMH);
 }
 
+void AP_OSD_Screen::draw_gcourse(uint8_t x, uint8_t y)
+{
+    AP_GPS & gps = AP::gps();
+    uint16_t cog = gps.ground_course();
+    backend->write(x, y, false, "%3d%c", cog, SYM_DEGR);
+}
+
+void AP_OSD_Screen::draw_throttle(uint8_t x, uint8_t y)
+{
+    uint8_t thr = SRV_Channels::get_output_scaled(SRV_Channel::k_throttle);
```

Comment:
this won't work for multicopters. 
I think we should add a set_throttle() method on the osd object which the vehicle code would call to fill in a value to use for throttle

Reply:
thanks tridge, will try and set groundcourse source to ahrs. 

adding the method for vehicle throttle is well beyond my scope though, would highly appreciate respective code corrections.

Reply:
Look at AP_Devo_Telem.h update_control_mode . it is defined in library and  called from vehicle classes.

Reply:
removed throttle item for now and changed gcourse to heading from ahrs source. thanks for your guidance sh83 and tridge.